CMakeLists.txt

```diff
@@ -44,19 +34,43 @@ if (ccache_binary)
 endif()
 
 # MPI package
-if(CHECKPOINT_BUILD_TESTS_WITH_MPI)
+if(checkpoint_mpi_enabled)
  include(cmake/load_mpi_package.cmake)
  set(
    CHECK_POINT_MPI_PROC 2 CACHE STRING
    "Set number of proc used by MPI for the tests. 2 is the default."
  )
 endif()
 
-message (STATUS "Checkpoint build tests: ${CHECKPOINT_BUILD_TESTS}")
-message (STATUS "Checkpoint build examples: ${CHECKPOINT_BUILD_EXAMPLES}")
+message (STATUS "Checkpoint build tests: ${checkpoint_tests_enabled}")
+message (STATUS "Checkpoint build examples: ${checkpoint_examples_enabled}")
 
 include(cmake/load_package.cmake)
 
+option(
+  checkpoint_doxygen_enabled "Build doxygen documentation for checkpoint" OFF
+)
+
+# Doxygen library
+if (${checkpoint_doxygen_enabled})
+  include(cmake/load_doxygen.cmake)
+endif()
+
+# Optionally enable address sanitizer library in build
+if (checkpoint_asan_enabled)
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" OR
+     CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR
+     (CMAKE_CXX_COMPILER_ID
+       STREQUAL
+       "GNU"
+       AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "4.8"))
```

Comment:
Do we really need to check that GCC is version > 4.8? We're most certainly using C++ features that would fail on older GCC anyway. I think the language flags setting elsewhere should fail to configure with a too-old compiler.

Reply:
Probably not, but I'll leave this in there just to be sure. I think checkpoint will compile with old compilers.